File: RiotGames/Dto/Match/PerksDto.cs
using Newtonsoft.Json;

namespace potg.RiotGames.Dto.Match;

public sealed record PerksDto
{
    [JsonProperty("statPerks")]
    public PerkStatsDto StatPerks { get; init; }

    [JsonProperty("styles")]
    public IReadOnlyCollection<PerkStyleDto> Styles { get; init; }
}

File: src/models/dto/riot/lol/match/perks.dto.ts
import { Field, ObjectType } from "@nestjs/graphql";
import { PerkStatsDto } from "./perkStats.dto";
import { PerkStyleDto } from "./perkStyle.dto";

@ObjectType()
export class PerksDto {
  @Field(() => PerkStatsDto)
  statPerks: PerkStatsDto;
  @Field((type) => [PerkStyleDto])
  styles: PerkStyleDto[];
}
